src/Umbraco.Core/Handlers/PublicAccessHandler.cs

```diff
@@ -7,32 +7,61 @@
 namespace Umbraco.Cms.Core.Handlers;
 
 public sealed class PublicAccessHandler :
+    INotificationAsyncHandler<MemberGroupSavingNotification>,
     INotificationHandler<MemberGroupSavedNotification>,
+    INotificationAsyncHandler<MemberGroupDeletingNotification>,
     INotificationHandler<MemberGroupDeletedNotification>
 {
     private readonly IPublicAccessService _publicAccessService;
+    private readonly IMemberGroupService _memberGroupService;
 
-    public PublicAccessHandler(IPublicAccessService publicAccessService) =>
-        _publicAccessService = publicAccessService ?? throw new ArgumentNullException(nameof(publicAccessService));
+    public PublicAccessHandler(IPublicAccessService publicAccessService, IMemberGroupService memberGroupService)
+    {
+        _publicAccessService = publicAccessService;
+        _memberGroupService = memberGroupService;
+    }
+
+    public async Task HandleAsync(MemberGroupSavingNotification notification, CancellationToken cancellationToken)
+        => await SaveStateAsync(notification.SavedEntities, notification);
+
+    public async Task HandleAsync(MemberGroupDeletingNotification notification, CancellationToken cancellationToken)
+        => await SaveStateAsync(notification.DeletedEntities, notification);
 
-    public void Handle(MemberGroupDeletedNotification notification) => Handle(notification.DeletedEntities);
+    public void Handle(MemberGroupDeletedNotification notification) => Handle(notification.DeletedEntities, notification);
+
+    public void Handle(MemberGroupSavedNotification notification) => Handle(notification.SavedEntities, notification);
+
+    private async Task SaveStateAsync(IEnumerable<IMemberGroup> affectedEntities, IStatefulNotification notification)
+    {
+        foreach (IMemberGroup memberGroup in affectedEntities)
+        {
+            // store the current group name in the notification state
+            var currentMemberGroupName = (await _memberGroupService.GetAsync(memberGroup.Key))?.Name;
+            if (currentMemberGroupName.IsNullOrWhiteSpace())
```

Comment:
We could consider to move the functionality into the service about gettting many, just to improve it in the future